db/cassandra/src/main/java/org/commonjava/indy/cassandra/data/CassandraStoreDataManager.java

```diff
@@ -0,0 +1,348 @@
+import org.commonjava.indy.db.common.AbstractStoreDataManager;
+import org.commonjava.indy.model.core.ArtifactStore;
+import org.commonjava.indy.model.core.Group;
+import org.commonjava.indy.model.core.HostedRepository;
+import org.commonjava.indy.model.core.RemoteRepository;
+import org.commonjava.indy.model.core.StoreKey;
+import org.commonjava.indy.model.core.StoreType;
+import org.commonjava.indy.model.core.io.IndyObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Alternative;
+import javax.inject.Inject;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@ApplicationScoped
+@Alternative
+public class CassandraStoreDataManager extends AbstractStoreDataManager
+{
+
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    @Inject
+    CassandraStoreQuery storeQuery;
+
+    @Inject
+    IndyObjectMapper objectMapper;
+
+    @Override
+    protected StoreEventDispatcher getStoreEventDispatcher()
+    {
+        return null;
+    }
+
+    @Override
+    protected ArtifactStore getArtifactStoreInternal( StoreKey key )
+    {
+        DtxArtifactStore dtxArtifactStore = storeQuery.getArtifactStore( key.getPackageType(), key.getType(), key.getName() );
+
+        return toArtifactStore( dtxArtifactStore );
+    }
+
+    @Override
+    protected ArtifactStore removeArtifactStoreInternal( StoreKey key )
+    {
+        DtxArtifactStore dtxArtifactStore = storeQuery.removeArtifactStore( key.getPackageType(), key.getType(), key.getName() );
+        return toArtifactStore( dtxArtifactStore );
+    }
+
+    @Override
+    public void clear( ChangeSummary summary ) throws IndyDataException
+    {
+
+    }
+
+    @Override
+    public Set<ArtifactStore> getAllArtifactStores()
+    {
+        Set<DtxArtifactStore> dtxArtifactStoreSet = storeQuery.getAllArtifactStores();
+        Set<ArtifactStore> artifactStoreSet = new HashSet<>(  );
+        dtxArtifactStoreSet.forEach( dtxArtifactStore -> {
+            artifactStoreSet.add( toArtifactStore( dtxArtifactStore ) );
+        } );
+        return artifactStoreSet;
+    }
+
+    @Override
+    public Map<StoreKey, ArtifactStore> getArtifactStoresByKey()
+    {
+        Map<StoreKey, ArtifactStore> ret = new HashMap<>();
+        Set<ArtifactStore> artifactStoreSet = getAllArtifactStores();
+        artifactStoreSet.forEach( store -> {
+            ret.put( store.getKey(), store );
+        } );
+        return ret;
+    }
+
+    @Override
+    public boolean hasArtifactStore( StoreKey key )
+    {
+        ArtifactStore artifactStore = getArtifactStoreInternal( key );
+        return artifactStore != null;
+    }
+
+    @Override
+    public boolean isStarted()
+    {
+        return false;
```

Comment:
If we don't use this, we should deprecate it, I guess. Otherwise, I would think we would always return true...